components/Chips/src/MDCChipField.m

```diff
@@ -0,0 +1,641 @@
+static const UIKeyboardType MDCChipFieldDefaultKeyboardType = UIKeyboardTypeEmailAddress;
+
+const CGFloat MDCChipFieldDefaultMinTextFieldWidth = 60.f;
+const UIEdgeInsets MDCChipFieldDefaultContentEdgeInsets = {
+    MDCChipFieldVerticalInset, MDCChipFieldHorizontalInset, MDCChipFieldVerticalInset,
+    MDCChipFieldHorizontalInset};
+
+@protocol MDCChipTextFieldDelegate <NSObject>
+
+- (void)textFieldShouldRespondToDeleteBackward:(UITextField *)textField;
+
+@end
+
+@interface MDCChipTextField : MDCTextField
+
+@property(nonatomic, weak) id<MDCChipTextFieldDelegate> deletionDelegate;
+
+@end
+
+@implementation MDCChipTextField
+
+#pragma mark UIKeyInput
+
+- (void)deleteBackward {
+  [super deleteBackward];
+  if (self.text.length == 0) {
+    [self.deletionDelegate textFieldShouldRespondToDeleteBackward:self];
+  }
+}
+
+#if MDC_CHIPFIELD_PRIVATE_API_BUG_FIX && !(defined(__IPHONE_8_3) && (__IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_8_3))
+
+// WARNING: This is a private method, see the warning in MDCChipField.h.
+// This is only compiled if you explicitly defined MDC_CHIPFIELD_PRIVATE_API_BUG_FIX yourself, and
+// you are targeting an iOS version less than 8.3.
+- (BOOL)keyboardInputShouldDelete:(UITextField *)textField {
+  BOOL shouldDelete = YES;
+  if ([UITextField instancesRespondToSelector:_cmd]) {
+    BOOL (*keyboardInputShouldDelete)(id, SEL, UITextField *) =
+        (BOOL (*)(id, SEL, UITextField *))[UITextField instanceMethodForSelector:_cmd];
+    if (keyboardInputShouldDelete) {
+      shouldDelete = keyboardInputShouldDelete(self, _cmd, textField);
+      NSOperatingSystemVersion minimumVersion = {8, 0, 0};
+      NSOperatingSystemVersion maximumVersion = {8, 3, 0};
+      NSProcessInfo *processInfo = [NSProcessInfo processInfo];
+      BOOL isIos8 = [processInfo isOperatingSystemAtLeastVersion:minimumVersion];
+      BOOL isLessThanIos8_3 = ![processInfo isOperatingSystemAtLeastVersion:maximumVersion];
+      if (![textField.text length] && isIos8 && isLessThanIos8_3) {
+        [self deleteBackward];
+      }
+    }
+  }
+  return shouldDelete;
+}
+
+#endif
+
+#pragma mark - UIAccessibility
+
+- (CGRect)accessibilityFrame {
+  CGRect frame = [super accessibilityFrame];
+  return CGRectMake(frame.origin.x + self.textInsets.left,
+                    frame.origin.y,
+                    frame.size.width - self.textInsets.left,
+                    frame.size.height);
+}
+
+@end
+
+@interface MDCChipField ()
+    <MDCChipTextFieldDelegate, MDCTextInputPositioningDelegate, UITextFieldDelegate>
+@end
+
+@implementation MDCChipField {
+  NSMutableArray<MDCChipView *> *_chips;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCChipFieldInit];
+
+    _chips = [NSMutableArray array];
+
+    MDCChipTextField *chipTextField = [[MDCChipTextField alloc] initWithFrame:self.bounds];
+    chipTextField.underline.hidden = YES;
```

Comment:
Can clients change the inset of the underline so the underline doesn't appear under chips? Maybe more of a MDCTextField question.

Reply:
I don't know... I'll have to find out.